app/views/portal/show.rb

```diff
@@ -455,152 +455,98 @@ def merge_values(fields, separator = ' ')
       collect_values(fields).join(separator)
     end
 
-    def data_section(data)
-      section_data = []
-      section_labels = []
+    def data_section_field_values(section)
+      fields = (section[:fields] || []).map do |field|
+        @document.fetch(field, [])
+      end
 
-      data[:sections].map do |section|
-        f_data = []
-        field_values = []
+      if section[:fields_then_fallback] && fields.present?
+        values = fields
+      else
+        values = [section[:collected]] + fields
+      end
 
-        if section[:collected]
-          f_data.push(* section[:collected])
-        end
-        if section[:fields]
-          # field_values = collect_values(section[:fields])
-          field_values = []
-          section[:fields].each do |field|
-            values = document.fetch(field, [])
-            if values.is_a? Array
-              values = values - field_values
+      values.flatten.compact.uniq
+    end
+
+    def data_section_field_subsection(section)
```

Comment:
Cyclomatic complexity for data_section_field_subsection is too high. [10/6]<br>Assignment Branch Condition size for data_section_field_subsection is too high. [43.98/15]<br>Method has too many lines. [33/10]<br>Perceived complexity for data_section_field_subsection is too high. [11/7]